CoreScriptsRoot/Modules/VR/LaserPointer.lua

```diff
@@ -0,0 +1,548 @@
+--LaserPointer.lua
+--Implements the visual part of the VR laser pointer and implements
+--VR teleporting and movement mechanics
+--Written by Kyle, September 2016
+local CoreGui = game.CoreGui
+local RobloxGui = CoreGui:WaitForChild("RobloxGui")
+local ContextActionService = game:GetService("ContextActionService")
+local Players = game:GetService("Players")
+local SoundService = game:GetService("SoundService")
+local PathfindingService = game:GetService("PathfindingService")
+local Utility = require(RobloxGui.Modules.Settings.Utility) --todo: use common utility module when it's done
+
+local LocalPlayer = Players.LocalPlayer
+
+--Pathfinding sort of works, but it's very slow and does not handle slopes very well.
+--Use at your own risk.
+local usePathfinding = false
+
+local gamma, invGamma = 2.2, 1/2.2
+local function fromLinearRGB(color)
+	return Color3.new(color.r ^ gamma, color.g ^ gamma, color.b ^ gamma)
+end
+local function toLinearRGB(color)
+	return Color3.new(color.r ^ invGamma, color.g ^ invGamma, color.b ^ invGamma)
+end
+
+local CONTAINER_FOLDER_NAME = "VRCorePanelParts"
+local function setPartInGame(part, inGame)
+	if not part then
+		return
+	end
+	if inGame and not part:IsDescendantOf(game) then
+		local container = workspace.CurrentCamera:FindFirstChild(CONTAINER_FOLDER_NAME)
+		if not container then
+			coroutine.wrap(function()
+				part.Parent = workspace.CurrentCamera:WaitForChild(CONTAINER_FOLDER_NAME)
```

Comment:
It doesn't feel great to put this part under the camera, we are putting a lot of core script stuff in the workspace again and again.